.github/workflows/release_artifacts.yaml

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build release artifacts
+
+on:
+    workflow_dispatch:
+        inputs:
+            releaseTag:
+                description: Release Tag
+                required: true
+
+jobs:
+    esp32:
+        name: ESP32
+        env:
+            BUILD_TYPE: esp32
+        runs-on: ubuntu-latest
+
+        container:
+            image: connectedhomeip/chip-build-esp32:0.4.18
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              with:
+                  submodules: true
+            - name: Build
+              run: scripts/examples/esp_echo_app.sh
+
+            - name: Upload artifact
+              run: |
+                 scripts/helpers/upload_release_asset.py                  \
+                    --github-repository project-chip/connectedhomeip      \
+                    --github-api-token "${{ secrets.GITHUB_TOKEN }}"      \
+                    --release-tag "${{ github.event.inputs.releaseTag }}" \
+                    --asset-path examples/all-clusters-app/esp32/build/m5stack-chip-all-clusters-app.elf \
+                    --asset-name esp32-m5stack-chip-all-clusters-app.elf
+    efr32:
+        name: EFR32
+        env:
+            EFR32_BOARD: BRD4161A
+            BUILD_TYPE: gn_efr32
+            BUILD_VERSION: 0.2.14
+
+        runs-on: ubuntu-latest
+
+        container:
+            image: connectedhomeip/chip-build-efr32:0.4.18
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+              with:
+                  submodules: true
+            - name: Build example EFR32 Lock App for BRD4161A
+              run:
+                  scripts/examples/gn_efr32_example.sh examples/lock-app/efr32/ out/lock_app_debug BRD4161A
```

Comment:
We should probably do a release build not a debug build here...

(the shell script needs extending, though. Needs to pass is_debug=false to GN)

Reply:
Ack.

Expect we will do refinement over a few more passes. Ideally I would like a release build as well as stand-alone flashing utilities, so that anyone getting an image can just flash the example.